### PR TITLE
Add Column Metadata

### PR DESCRIPTION
**Pull request by MarcelHillmann**
For my testcases I need to mock the rows.ColumnTypes() function

### CODE REVIEW

File: rows.go [hunk 1]
```diff
@@ -49,6 +50,31 @@ func (rs *rowSets) Next(dest []driver.Value) error {
 	return r.nextErr[r.pos-1]
 }
 
+// ColumnTypeLength is defined from driver.RowColumnTypeLength
+func (rs *rowSets) ColumnTypeLength(index int) (length int64, ok bool) {
+	return rs.sets[0].def[index].length, false
```
Comment (l3pp4rd) [inline rows.go]: there may be more than one ResultSet, use `rs.pos` instead of `0` index.

File: rows.go [hunk 2]
```diff
@@ -80,24 +106,63 @@ func (rs *rowSets) empty() bool {
...
 	return &Rows{
 		cols:      columns,
+		def:       definition,
+		nextErr:   make(map[int]error),
+		converter: driver.DefaultParameterConverter,
+	}
+}
+
+// NewRowsWithColumnDefiniton see PR-152
+func NewRowsWithColumnDefiniton(columns ...*Column) *Rows {
```
Comment (l3pp4rd) [inline rows.go]: sqlmock interface can also [initialize Rows](https://github.com/DATA-DOG/go-sqlmock/blob/master/sqlmock.go#L76) with an user defined column converter. The same method should be there too.

File: rows_go18_test.go [hunk 3]
```diff
@@ -90,3 +91,56 @@ func TestQueryMultiRows(t *testing.T) {
 		t.Errorf("there were unfulfilled expectations: %s", err)
 	}
 }
+
+func TestNewColumnWithDefinition(t *testing.T) {
```
Comment (l3pp4rd) [inline rows_go18_test.go]: Also add a test for two result sets

File: rows.go [hunk 4]
```diff
@@ -80,24 +106,63 @@ func (rs *rowSets) empty() bool {
 	return true
 }
 
+// Column is a mocked column Metadate for rows.ColumnTypes()
+type Column struct {
```
Comment (l3pp4rd) [inline rows.go]: maybe in that case, instead of having exported construction methods, just export the struct fields instead?

**Comment by l3pp4rd** [comment]
Hi, please rebase an squash the commits also. I'll check a bit more later related to column types and definition.

**Comment by ain-valtin** [comment]
Ping @l3pp4rd @MarcelHillmann any chance merging it / releasing new version? I'd love to have a `ColumnTypes()` which returns "good data"...

**Comment by l3pp4rd** [comment]
will move to #187 but need to review it closely, do not have time now, sorry about that guys. you can temporary use a fork

**Pull request closed without merging**